tensornetwork/matrixproductstates/mpo_test.py

```diff
@@ -5,7 +5,26 @@
 import torch
 from tensornetwork.backends import backend_factory
 #pylint: disable=line-too-long
-from tensornetwork.matrixproductstates.mpo import FiniteMPO, BaseMPO, InfiniteMPO
+from tensornetwork.matrixproductstates.mpo import (FiniteMPO,
+                                                   BaseMPO,
+                                                   InfiniteMPO,
+                                                   FiniteFreeFermion2D)
+from tensornetwork.matrixproductstates.finite_mps import FiniteMPS
+from tensornetwork.matrixproductstates.dmrg import FiniteDMRG
+
+
+def adjacency(N1, N2):
+  neighbors = {}
+  mat = np.arange(N1 * N2).reshape(N1, N2)
+  for n in range(N1 * N2):
+    x, y = np.divmod(n, N2)
+    if n not in neighbors:
+      neighbors[n] = []
+    if y < N2 - 1:
+      neighbors[n].append(mat[x, y + 1])
+    if x > 0:
+      neighbors[n].append(mat[x - 1, y])
```

Comment:
This only needs to find forward adjacency?

Reply:
that's right